bchrpc/server.go

```diff
@@ -1194,19 +1213,30 @@ func (s *GrpcServer) SubscribeTransactionStream(stream pb.Bchrpc_SubscribeTransa
 						continue
 					}
 
-					header := block.MsgBlock().Header
+					toSend := &pb.TransactionNotification{}
+					toSend.Type = pb.TransactionNotification_CONFIRMED
 
-					respTx := marshalTransaction(tx, s.chain.BestSnapshot().Height-block.Height(), &header, block.Height(), s.chainParams)
-					if s.txIndex != nil {
-						if err := s.setInputMetadata(respTx); err != nil {
-							return err
+					if serializeTx {
+						var buf bytes.Buffer
+						if err := tx.MsgTx().BchEncode(&buf, wire.ProtocolVersion, wire.BaseEncoding); err != nil {
+							return status.Error(codes.Internal, "error serializing transaction")
 						}
-					}
-					toSend := &pb.TransactionNotification{
-						Type: pb.TransactionNotification_CONFIRMED,
-						Transaction: &pb.TransactionNotification_ConfirmedTransaction{
+						toSend.Transaction = &pb.TransactionNotification_SerializedTransaction{
+							SerializedTransaction: buf.Bytes(),
+						}
+
+					} else {
+						header := block.MsgBlock().Header
+
+						respTx := marshalTransaction(tx, s.chain.BestSnapshot().Height-block.Height(), &header, block.Height(), s.chainParams)
+						if s.txIndex != nil {
+							if err := s.setInputMetadata(respTx); err != nil {
```

Comment:
Comment for `setInputMetadata` says it is used when txIndex is NOT available? But this seems to only do it when it is... ?

Reply:
I should probably rename that helper function to avoid confusion, since there are 2. 
A method on GrpcServer:
`func (s *GrpcServer) setInputMetadata(tx *pb.Transaction) error`

And a helper function:
`func setInputMetadata(respTx *pb.Transaction, txDesc *rpcEventTxAccepted, view *blockchain.UtxoViewpoint, chainParams *chaincfg.Params)`

The former is used when `s.txIndex != nil` the latter when `s.txIndex == nil`

Reply:
Thanks for the clarification.